iib/web/models.py

```diff
@@ -288,13 +334,17 @@ def _state_to_json(state):
             'binary_image_resolved': getattr(
                 self.binary_image_resolved, 'pull_specification', None
             ),
-            'bundles': [bundle.pull_specification for bundle in self.bundles],
             'from_index': getattr(self.from_index, 'pull_specification', None),
             'from_index_resolved': getattr(self.from_index_resolved, 'pull_specification', None),
             'index_image': getattr(self.index_image, 'pull_specification', None),
             'user': getattr(self.user, 'username', None),
         }
 
+        if self.type == RequestTypeMapping.__members__['add'].value:
+            rv['bundles'] = [bundle.pull_specification for bundle in self.bundles]
+        else:
+            rv['operators'] = [operator.name for operator in self.operators]
```

Comment:
Just a question here. What is the format of operator in relation to bundle? So bundle is `<host>/<space>/<name>:<tag>` and operator in that context is `<space>/<name>:<tag>` ?
I'm thinking if it would be possible to return operators when adding bundles as well. It makes sense to me to gather this logic rather here (mapping bundles -> operators) than in pub

Reply:
The purpose of storing the operators in the database is to audit what the original request is and to replay the request if it fails. IIB will not discover what the relationship is between bundles and operators.